src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -162,6 +162,9 @@ data AgentClient = AgentClient
     connMsgsQueued :: TMap ConnId Bool,
     smpQueueMsgQueues :: TMap (SMPServer, SMP.SenderId) (TQueue InternalId),
     smpQueueMsgDeliveries :: TMap (SMPServer, SMP.SenderId) (Async ()),
+    serverCmdsQueued :: TMap (Maybe SMPServer) Bool,
+    asyncCmdQueues :: TMap (Maybe SMPServer) (TQueue AsyncCmdId),
+    asyncCmdProcesses :: TMap (Maybe SMPServer) (Async ()),
```

Comment:
```suggestion
    connCmdsQueued :: TMap (Maybe ConnId) Bool,
    connCmdQueues :: TMap ConnId (TQueue AsyncCmdId),
    srvCmdQueues :: TMap (Maybe SMPServer) (TQueue AsyncCmdId)
    connCmdAsync :: TMap ConnId (Async ()),
    srvCmdAsync :: TMap (Maybe SMPServer) (Async ()),
```

I think it could be something like this, where the process for a given connection just puts command in the server queue.

But maybe it would work as is, it's just past commands will be loaded on subscription, and the new ones are related to a subscribed connection anyway...